### ISSUE TITLE
Responsive design  в статичном HTML

### ISSUE THREAD

**Issue by Guria** [bem/bem-forum-content-ru#263]
Давно хочу спросить: как, используя `bem-core` и `bem-components` собрать статичный html, который можно будет использовать на любой платформе. Т.е. я хочу, например, чтобы блоки `page` и `ua` были собраны из  всех слоёв переопределения (`touch`, `desktop`).

Дело в том, что библиотеки [не поддерживают работу этих слоёв одновременно](https://github.com/bem/bem-core/issues/835), и более того - не собираются.

Как быть, что делать?

**Comment by apsavin** [comment]
Мне кажется, единственный разумный подход - это вообще отказаться от деления тач/не тач. Не очень понятно, зачем это деление нужно, учитывая кучу гибридных устройств.

Для меня это означает, что на уровне проекта нужно допиливать десктопный уровень до такого состояния, чтобы он поддерживал тач-события. Это не очень сложно, если использовать полифил для pointer events.

**Comment by Guria** [comment]
Т.е. на уровне проекта, самостоятельно реализовывать, то что уже описано в библиотеке `bem-core` на уровне `touch`?

**Comment by tadatuta** [comment]
@Guria мы постарались унести в `common` то, что нужно, чтобы блоки из `desktop` заработали на тачах. Там, например, в `bem-core` туда попала реализация поинтер-ивентов. Контролы из `bem-components` по задумке полноценно работают на телефонах. Если это не так, то можно заводить баги, мы будем чинить.

**Comment by Guria** [comment]
Меня, в основном, интересовали:
https://github.com/bem/bem-core/blob/v2/touch.blocks/page/page.bh.js
https://github.com/bem/bem-core/blob/v2/touch.blocks/page/__icon/page__icon.bh.js
Можно их копирнуть на проект, конечно. Но стоит ли?
Не исключаю, что проблему я выдумал.

**Comment by apsavin** [comment]
@Guria да, возможно, я "страшно" описал, но на самом деле "самостоятельно реализовывать, то что уже описано в библиотеке bem-core на уровне touch" не приходится, конечно же.

**Comment by tadatuta** [comment]
@Guria `page` кажется, тянет на issue для обсуждения. Заведешь?

**Comment by Guria** [comment]
@tadatuta ОК

**Comment by Guria** [comment]
> Контролы из bem-components по задумке полноценно работают на телефонах. Если это не так, то можно заводить баги, мы будем чинить.

Можно вернуться к вопросу:
> Мне кажется, единственный разумный подход - это вообще отказаться от деления тач/не тач. Не очень понятно, зачем это деление нужно, учитывая кучу гибридных устройств.

**Comment by Guria** [comment]
В частности, `page` и `ua`, можно разнести по модификаторам и устранить лишний слой в `bem-core`.

**Comment by veged** [comment]
все гибридные устройства должны нормально работать с desktop уровнем

**Comment by Guria** [comment]
Разобрался с circular dependencies и предложил решение:
https://github.com/bem/bem-core/issues/835#issuecomment-77110786
